In [29]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns

In [30]:
from google.colab import files
files.upload()

Saving Sales_add.csv to Sales_add (1).csv


{'Sales_add.csv': b'Month,Region,Manager,Sales_before_digital_add(in $),Sales_After_digital_add(in $)\r\nMonth-1,Region - A,Manager - A,132921,270390\r\nMonth-2,Region - A,Manager - C,149559,223334\r\nMonth-3,Region - B,Manager - A,146278,244243\r\nMonth-4,Region - B,Manager - B,152167,231808\r\nMonth-5,Region - C,Manager - B,159525,258402\r\nMonth-6,Region - A,Manager - B,137163,256948\r\nMonth-7,Region - C,Manager - C,130625,222106\r\nMonth-8,Region - A,Manager - A,131140,230637\r\nMonth-9,Region - B,Manager - C,171259,226261\r\nMonth-10,Region - C,Manager - B,141956,193735\r\nMonth-11,Region - B,Manager - C,159339,203175\r\nMonth-12,Region - A,Manager - A,178939,276279\r\nMonth-13,Region - B,Manager - A,145062,205180\r\nMonth-14,Region - A,Manager - A,151514,253131\r\nMonth-15,Region - C,Manager - A,147463,229336\r\nMonth-16,Region - A,Manager - B,177195,187305\r\nMonth-17,Region - A,Manager - C,140862,234908\r\nMonth-18,Region - C,Manager - B,167996,191517\r\nMonth-19,Region - B,Ma

In [31]:
df=pd.read_csv('Sales_add.csv')

In [32]:
df.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

#The company wishes to clarify whether there is any increase in sales after stepping into digital marketing
Null hypothes H0: sales after digital marketing <= sales before digital marketing

Alternative hypothes H1: sales after digital marketing > sales before digital marketing

In [33]:
before=df['Sales_before_digital_add(in $)']
before

0     132921
1     149559
2     146278
3     152167
4     159525
5     137163
6     130625
7     131140
8     171259
9     141956
10    159339
11    178939
12    145062
13    151514
14    147463
15    177195
16    140862
17    167996
18    132135
19    152493
20    147425
21    130263
Name: Sales_before_digital_add(in $), dtype: int64

In [34]:
after=df['Sales_After_digital_add(in $)']
after

0     270390
1     223334
2     244243
3     231808
4     258402
5     256948
6     222106
7     230637
8     226261
9     193735
10    203175
11    276279
12    205180
13    253131
14    229336
15    187305
16    234908
17    191517
18    227040
19    212579
20    263388
21    243020
Name: Sales_After_digital_add(in $), dtype: int64

In [35]:
from scipy.stats import chi2_contingency
t_stat,p_value=stats.ttest_ind(after,before)
print('The value t-statistics is: ',t_stat)
print('The value p-value is: ',p_value)

The value t-statistics is:  12.995084451110875
The value p-value is:  2.614368006904645e-16


**Assume alpha value = 0.05, from results obtained above p-value is lesser than alpha value. So reject null hypothesis. Thus, we can conclude there is increase in sales after digital marketing**

#The company needs to check whether there is any dependency between the features “Region” and “Manager”.
Null hypothesis H0: Region and Manager are independent

Alternative hypothesis H1: Region and Manager are dependent

In [36]:
field1 = df['Region']
field2 = df['Manager']

In [37]:
myCrosstable = pd.crosstab(field1,field2)
myCrosstable

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [38]:
from scipy.stats import chi2_contingency

In [39]:
chiVal, pVal, df,exp = chi2_contingency(myCrosstable)

In [40]:
chiVal,pVal, df,exp

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

**Assume alpha value=0.05, p-value=0.549, Thus p-value>alpha-value. Accept Null Hypothesis. Thus, Region and Manager are independent.**